martin/src/cog/errors.rs

```diff
@@ -0,0 +1,49 @@
+use std::path::PathBuf;
+
+use png::EncodingError;
+use tiff::TiffError;
+
+#[derive(thiserror::Error, Debug)]
+pub enum CogError {
+    #[error("Couldn't decoded {1} as tiff file: {0}")]
+    InvalidTifFile(TiffError, PathBuf),
+
+    #[error("Requested zoom level:{0} from file {1} is  out of range, the zoom level is from {2} to {3}")]
```

Comment:
unsure if this is better, but the current error message seems a bit off

```suggestion
    #[error("Requested zoom level {0} from file {1} is out of range. Possible zoom levels are {2} to {3}")]
```

Reply:
Thx! Like your suggestion.